scene/gui/scroll_container.cpp

```diff
@@ -57,9 +57,13 @@ Size2 ScrollContainer::get_minimum_size() const {
 
 	if (horizontal_scroll_mode == SCROLL_MODE_DISABLED) {
 		min_size.x = MAX(min_size.x, largest_child_min_size.x);
+	} else if (max_size.x > 0.0) {
+		min_size.x = MIN(max_size.x, MAX(min_size.x, largest_child_min_size.x));
```

Comment:
`min_size.x` is always `0` at this point. Also `largest_child_min_size.x` can't be less than `0` either. I wonder what the MAX is for. It's unnecessary above too.